### PR TITLE
Migrate map_filter function to use MapBlock

### PR DESCRIPTION
**Pull request by haozhun**

### CODE REVIEW

**Comment by haozhun** [comment]
I noticed this after I merged ae09a91b. This PR needs to be reviewed before release otherwise it would be a regression.

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MapFilterFunction.java [hunk 1]
```diff
@@ -132,18 +142,21 @@ private static MethodHandle generateFilter(Type keyType, Type valueType, Type ma
...
-                ImmutableList.of(block, function));
+                ImmutableList.of(state, block, function));
 
         BytecodeBlock body = method.getBody();
         Scope scope = method.getScope();
         Variable positionCount = scope.declareVariable(int.class, "positionCount");
         Variable position = scope.declareVariable(int.class, "position");
+        Variable pageBuilder = scope.declareVariable(PageBuilder.class, "pageBuilder");
+        Variable mapBlockBuilder = scope.declareVariable(BlockBuilder.class, "mapBlockBuilder");
         Variable blockBuilder = scope.declareVariable(BlockBuilder.class, "blockBuilder");
```
Comment (wenleix) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/MapFilterFunction.java]: I think usually you like to use `singleMapBlockWriter` here as the variable name :)

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/MapFilterFunction.java [hunk 2]
```diff
@@ -195,9 +209,19 @@ private static MethodHandle generateFilter(Type keyType, Type valueType, Type ma
                                         .append(keySqlType.invoke("appendTo", void.class, block, position, blockBuilder))
                                         .append(valueSqlType.invoke("appendTo", void.class, block, add(position, constantInt(1)), blockBuilder))))));
 
-        body.append(blockBuilder.invoke("build", Block.class).ret());
+        body.append(mapBlockBuilder
+                .invoke("closeEntry", BlockBuilder.class)
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/MapFilterFunction.java]: Do we need to worry about entry not being closed if there is an exception?

Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/MapFilterFunction.java]: We do. `map_filter` doesn't throw exceptions. However, you're right in that the evaluation of the lambda expression could throw exceptions. This issue also applies to other map functions. This need to be addressed separately.

**Pull request merged**